core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -674,6 +677,18 @@ private[spark] class Executor(
     }
   }
 
+  private def setMDCForTask(taskDescription: TaskDescription): Unit = {
+    val properties = taskDescription.properties
+
+    org.slf4j.MDC.put("appId", properties.getProperty("spark.app.id"))
```

Comment:
I can't find the code that sets app name and id to the task properties, can you point me to the corresponding code?

Reply:
SparkContext.scala:574

Reply:
here we set it after reading it from the network...as far as I understand the code.

Reply:
How can we set it without carrying the app name/id somewhere?

Reply:
few lines above you have the encode logic

Reply:
Sorry I still can't find it in the `encode` method
https://github.com/apache/spark/blob/559fe098f666065f5411ec373dc466b260f6f87d/core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala#L85-L121

Did I msis somethinng?

Reply:
in the TaskSetManager we create:
```scala
new TaskDescription(
          taskId,
          attemptNum,
          execId,
          taskName,
          index,
          task.partitionId,
          addedFiles,
          addedJars,
          task.localProperties,
          extraResources,
          serializedTask)
      }
```
so we take the task.localProperties and pass it to the `TaskDescription`

Reply:
And where do we set app id/name to `task.localProperties`? Do users need to do it manually?

Reply:
it comes from DAGScheduler